backend/src/test/java/codezap/tag/repository/TagJpaRepositoryTest.java

```diff
@@ -0,0 +1,133 @@
+package codezap.tag.repository;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+import java.util.List;
+import java.util.Optional;
+
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+
+import codezap.global.exception.CodeZapException;
+import codezap.global.repository.JpaRepositoryTest;
+import codezap.tag.domain.Tag;
+
+@JpaRepositoryTest
+class TagJpaRepositoryTest {
+
+    private final TagRepository tagRepository;
+
+    @Autowired
+    TagJpaRepositoryTest(TagRepository tagRepository) {
+        this.tagRepository = tagRepository;
+    }
+
+    @Nested
+    @DisplayName("id로 태그 조회")
+    class fetchById {
+        @Test
+        @DisplayName("성공 : id로 태그를 알아낼 수 있다.")
+        void fetchByIdSuccess() {
+            Tag tag = tagRepository.save(new Tag("tag"));
+
+            Tag actual = tagRepository.fetchById(tag.getId());
+
+            assertThat(actual).isEqualTo(tag);
+        }
+
+        @Test
+        @DisplayName("실패 : 존재하지 않는 id인 경우 에러가 발생한다.")
+        void fetchByIdFailByNotExistsId() {
+            long id = 100;
+
+            assertThatThrownBy(() -> tagRepository.fetchById(id))
+                    .isInstanceOf(CodeZapException.class)
+                    .hasMessage("식별자 " + id + "에 해당하는 태그가 존재하지 않습니다.");
+        }
+    }
+
+    @Nested
+    @DisplayName("태그명으로 태그 조회")
+    class fetchByName {
+        @Test
+        @DisplayName("성공 : 태그명으로 태그를 알아낼 수 있다.")
+        void fetchByNameSuccess() {
+            Tag tag = tagRepository.save(new Tag("tag"));
+
+            Tag actual = tagRepository.fetchByName(tag.getName());
+
+            assertThat(actual).isEqualTo(tag);
+        }
+
+        @Test
+        @DisplayName("실패 : 존재하지 않는 태그명인 경우 에러가 발생한다.")
+        void fetchByNameFailByNotExistsId() {
+            String name = "태그";
+
+            assertThatThrownBy(() -> tagRepository.fetchByName(name))
+                    .isInstanceOf(CodeZapException.class)
+                    .hasMessage("이름이 " + name + "인 태그는 존재하지 않습니다.");
+        }
```

Comment:
```suggestion
        @Test
        @DisplayName("실패 : 존재하지 않는 태그명인 경우 에러가 발생한다.")
        void fetchByNameFailByNotExistsId() {
            String notExistTagName = "없는태그";

            assertThatThrownBy(() -> tagRepository.fetchByName(notExistTagName))
                    .isInstanceOf(CodeZapException.class)
                    .hasMessage("이름이 " + notExistTagName + "인 태그는 존재하지 않습니다.");
        }
```
[제안]
다른 리뷰와 마찬가지로 변수 명, 변수 값을 더 명확히 변경해 보았어요.